ir/src/main/scala/org/scalajs/core/ir/Serializers.scala

```diff
@@ -634,11 +646,36 @@ object Serializers {
     }
 
     def readTree(): Tree = {
+      val pos = readPosition()
+      readTreeFromTag(input.readByte())(pos)
+    }
+
+    def readOptTree(): Option[Tree] = {
+      if (useHacks068) {
+        readTreeAllowEmpty()
+      } else {
+        import input._
+        if (readBoolean()) {
+          Some(readTree())
+        } else {
+          None
+        }
+      }
+    }
+
+    private def readTreeAllowEmpty(): Option[Tree] = {
+      require(useHacks068)
+      val pos = readPosition()
+      val tag = input.readByte()
+      if (tag == TagEmptyTree) None
+      else Some(readTreeFromTag(tag)(pos))
```

Comment:
I wonder whether it wouldn't be a good idea to keep this old encoding even for the new format. i.e., `TagEmptyTree` for `None` and other tags for `Some(_)`. This would save one byte every time.